chore: fix C lint errors (issue #8156)

GeoDaoyu opened the pull request:
Resolves #8156.

## Description

> What is the purpose of this pull request?

This pull request:

-   fix C lint errors

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #8156

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

GeoDaoyu committed:
chore: fix C lint errors (issue #8156)

kgryte commented on lib/node_modules/@stdlib/ndarray/base/assert/is-single-segment-compatible/src/main.c:
```diff
@@ -49,7 +49,7 @@
 * // returns 1
 */
-int8_t stdlib_ndarray_is_single_segment_compatible( enum STDLIB_NDARRAY_DTYPE dtype, int64_t ndims, int64_t *shape, int64_t *strides, int64_t offset ) {
+int8_t stdlib_ndarray_is_single_segment_compatible( enum STDLIB_NDARRAY_DTYPE dtype, int64_t ndims, int64_t *shape, const int64_t *strides, int64_t offset ) {
```
@GeoDaoyu In general, for changes such as this, you also need to update the header file and the README (if the function signature is documented there).

kgryte committed:
refactor: add `const` qualifiers

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM after making some more general changes.
